crates/driver/src/logic/eth.rs

```diff
@@ -7,5 +9,82 @@ pub struct NetworkName(pub String);
 /// Chain ID as defined by EIP-155.
 ///
 /// https://eips.ethereum.org/EIPS/eip-155
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy)]
 pub struct ChainId(pub u64);
+
+/// Gas amount.
+#[derive(Debug, Clone, Copy)]
+pub struct Gas(pub U256);
+
+/// Gas price.
+/// TODO This will probably need to be different, autopilot uses GasPrice1559
+#[derive(Debug, Clone, Copy)]
+pub struct GasPrice(pub U256);
+
+/// An EIP-2930 access list.
+///
+/// https://eips.ethereum.org/EIPS/eip-2930
+#[derive(Debug)]
+pub struct AccessList(pub web3::types::AccessList);
+
+impl AccessList {
+    pub fn merge(_other: Self) -> Self {
+        todo!()
+    }
+}
+
+/// The results of an Ethereum transaction simulation.
+#[derive(Debug)]
+pub struct Simulation {
+    pub gas: Gas,
+    pub access_list: AccessList,
+}
+
+/// An address. Can be an EOA or a smart contract address.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Address(pub H160);
```

Comment:
Personally, I am a huge fan of these new-types. I believe they simultaneously help in two ways:

1. convey code intent more clearly. For example, in the code below, you very clearly know you are talking about buy and sell tokens and not amounts because of their type:
    ```rust
    let sell: Token;
    let buy: Token;
    ```
    
2. compile-time hints for hard to catch bugs. For example, a pretty innocuous:
    ```rust
    async fn get_balance(owner: Address, token: Address) -> Result<U256>;
    ```

    Now, at a glance, without looking back up to the signature, it is not obvious which one of these is correct:
    ```rust
    get_balance(my_wallet_address, cow_token).await;
    get_balance(cow_token, my_wallet_address).await;
    ```

The downside to me is additional boilerplate, but I think it is a very small price to pay in the long run.